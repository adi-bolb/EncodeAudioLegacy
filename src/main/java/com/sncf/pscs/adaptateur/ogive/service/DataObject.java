package com.sncf.pscs.adaptateur.ogive.service;

public interface DataObject {

	String getString(String arg0);

	boolean getBoolean(String key);

}
