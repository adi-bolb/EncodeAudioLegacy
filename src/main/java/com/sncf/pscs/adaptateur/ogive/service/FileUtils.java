package com.sncf.pscs.adaptateur.ogive.service;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.nio.channels.FileChannel;

public class FileUtils {

	/**
	 * Post a file. Create it and write its content.
	 * 
	 * @param localServerFolder
	 *            the url where to post the file
	 * @param fileName
	 *            the name of the file
	 * @param content
	 *            the content of the file
	 * @throws IOException 
	 */
	public static void postFile(final String localServerFolder, final String fileName, final byte[] content) throws IOException
	{
		FileOutputStream fos = null;
		try
		{
			final File localFile = createFile(localServerFolder, fileName);
			fos = new FileOutputStream(localFile);
			fos.write(content);
		}
		finally
		{
			if (fos != null)
			{
				try
				{
					fos.close();
				}
				catch (IOException e)
				{

				}
			}
		}
	}

	/**
	 * Copy a file to an other with a given name
	 * 
	 * @param source
	 * @param dest
	 * @throws FileNotFoundException
	 *             , IOException
	 * @throws IOException
	 */
	public static void copyFile(final String source, final String dest) throws FileNotFoundException, IOException
	{
		FileChannel channelIn = null;
		FileChannel channelOut = null;

		try
		{
			channelIn = new FileInputStream(source).getChannel();
			channelOut = new FileOutputStream(dest).getChannel();

			channelIn.transferTo(0, channelIn.size(), channelOut);
		}
		finally
		{
			if (channelIn != null)
			{
				channelIn.close();
			}
			if (channelOut != null)
			{
				channelOut.close();
			}
		}
	}
	/**
	 * Create a file.
	 * 
	 * @param path
	 *            the path of the file
	 * @param name
	 *            the name of the file
	 * @return an object of type {@link File} representing the file to create
	 * @throws IOException
	 */
	public static File createFile(final String path, final String name) throws IOException
	{
		final File file = new File(path + "/" + name);
		file.createNewFile();
		return file;
	}
	
	/**
	 * Check if a path is correct, if not, correct it.
	 * 
	 * @param path
	 *            the path to check.
	 * @return the path
	 */
	public static String checkPath(final String path)
	{

		String newPath = path;
		// Check the path is ended by '/' or '\'
		if (newPath != null && !newPath.isEmpty())
		{
			final char lastChar = newPath.charAt(newPath.length() - 1);
			if (lastChar != '/' && lastChar != '\\')
			{
				newPath = newPath.concat("/");
			}
		}
		else
		{
			newPath = "/";
		}
		return newPath;
	}
	
	/**
	 * Return true if file exist.
	 * 
	 * @param filePath
	 * @return boolean
	 */
	public static boolean isFileExist(final String filePath)
	{
		return new File(filePath).exists();
	}
}
