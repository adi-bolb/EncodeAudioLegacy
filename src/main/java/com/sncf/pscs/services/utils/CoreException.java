package com.sncf.pscs.services.utils;

/**
 * @author PAG
 *
 */
public class CoreException extends Exception{

	private static final long serialVersionUID = 563728075825478349L;

	/**
	 * @param message
	 */
	public CoreException (final String message) {
		super(message);
	}

	/**
	 * @param exeption
	 */
	public CoreException (final Exception exeption) {
		super(exeption);
	}

	/**
	 * @param message
	 * @param exeption
	 */
	public CoreException (final String message, final Exception exeption) {
		super(message, exeption);
	}
	
}
