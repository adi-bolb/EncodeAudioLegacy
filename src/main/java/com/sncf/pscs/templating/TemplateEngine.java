package com.sncf.pscs.templating;

import java.util.Map;

public interface TemplateEngine {

	String compile(String template, Map<String, String> vals);

}
