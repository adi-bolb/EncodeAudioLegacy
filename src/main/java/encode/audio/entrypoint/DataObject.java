package encode.audio.entrypoint;

public interface DataObject {

	String getString(String arg0);

	boolean getBoolean(String key);

}
