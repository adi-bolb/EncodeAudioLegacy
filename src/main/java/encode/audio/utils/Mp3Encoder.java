package encode.audio.utils;

import java.io.IOException;

public class Mp3Encoder {

	public static int launchMp3Exec(String cmd) throws IOException,
			InterruptedException {
		Process p1 = Runtime.getRuntime().exec(cmd);
		p1.getOutputStream().close();
		p1.getInputStream().close();
		p1.getErrorStream().close();
		p1.waitFor();
	
		// Exï¿½cution de la commande d'encodage audio
		//LOG.info("Fin de l'encodage du fichier audio, code retour de la commande :" + p.exitValue());
		int exitValue = p1.exitValue();
		return exitValue;
	}

}
