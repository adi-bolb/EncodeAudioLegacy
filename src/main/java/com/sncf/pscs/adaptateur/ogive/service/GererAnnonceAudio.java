package com.sncf.pscs.adaptateur.ogive.service;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

import pscs.tmlg.flux.FluxTmlg;
import pscs.tmlg.iflux.IAnnonceAudioTmlg;
import pscs.tmlg.iflux.IFluxTmlg;
import pscs.tmlg.utils.ObixTmlgExeption;

import com.sncf.pscs.services.utils.AudioFile;
import com.sncf.pscs.services.utils.CoreException;
import com.sncf.pscs.services.utils.DummyLogService;
import com.sncf.pscs.services.utils.HttpRequestFactory;
import com.sncf.pscs.services.utils.LogService;
import com.sncf.pscs.services.utils.PscsCoreUtil;

public class GererAnnonceAudio{
	
	private Boolean download = false;
	private Boolean encode = false;
	
	private LogService logger = new DummyLogService(); 
	private DataObject configAudio;
	private DataObject configHttp;
	
	/** The key to get the corresponding in DB. */
	private final static String KEY_CONF_URL_AUDIO_FOLDER = "AUDIO.ANNOUNCE.FOLDER.URL";
	
	public String demanderFlux(String message, DataObject configAudioTmp, DataObject configHttpTmp) throws PscsTechnicalException{
		try{
			this.configAudio = configAudioTmp;
			this.configHttp = configHttpTmp;
			IFluxTmlg flux;
			flux = new FluxTmlg(message);
			
			AudioFile newAudioFile = null;
			IAnnonceAudioTmlg annonceAudio = flux.getBody().getInfoVoyageur().getAnnonceAudio();

			// Traitement du message d'annonce audio
			newAudioFile = traiterAnnonceAudio(flux, annonceAudio);

			// Avertissement du bord d'une annonce audio
			flux = envoiAnnonceAudio(flux, newAudioFile);
			return flux.toString();
			
		}catch(CoreException ex){
 
			logger.log(LogService.LOG_ERROR,  ex.getMessage());
			if(!download){
 
				logger.log(LogService.LOG_ERROR,  ex.getMessage());
				throw new PscsTechnicalException(PscsCoreUtil.ERREUR_HTTP_DOWNLOAD, ex);
			}else if(!encode){	
 
				logger.log(LogService.LOG_ERROR,  ex.getMessage());
				throw new PscsTechnicalException(PscsCoreUtil.ERREUR_AUDIO_ENCODING, ex);
			}else{
 
				logger.log(LogService.LOG_ERROR,  ex.getMessage());
				throw new PscsTechnicalException(PscsCoreUtil.ERREUR_HTTP_UPLOAD, ex);
			}
		}catch (ObixTmlgExeption e) {
 
			logger.log(LogService.LOG_ERROR,  e.getMessage());
			throw new PscsTechnicalException(PscsCoreUtil.ERREUR_FORMAT_OBIX);
		} catch (Exception e) {
			logger.log(LogService.LOG_ERROR,  e.getMessage());
			throw new PscsTechnicalException(PscsCoreUtil.ERREUR_HTTP_CONFIG, e);
		}
	}
	
	
	/**
	 * Traitement optimise d'une annonce audio en verifiant si les fichiers audio ne sont pas deje accessible sur le serveur HTTPS
	 * Attention, les fichiers une fois telecharges ne sont pas supprimes en local
	 * @param flux
	 * @param annonceAudio
	 * @return AudioFile
	 * @throws CoreException 
	 * */
	public AudioFile traiterAnnonceAudio(IFluxTmlg flux, IAnnonceAudioTmlg annonceAudio) throws CoreException, Exception{
		
		String fileName = annonceAudio.getFileName();
		String fileUrl = annonceAudio.getUrl();
		String audioTemporalyPath = configHttp.getString("audio_temp_path");
		String filePath = audioTemporalyPath+fileName;
		
		AudioFile newAudioFile = PscsCoreUtil.simulateEncodedAudioFileProperties(fileName, configAudio.getString("final_audio_file_extension"));
		String encodedFilename = newAudioFile.getName();

		//Get the folder where to check or post the audio file
		String localServerFolder = getLocalServerFolder();
		
		if(verifAudioFileExistsOnServer(localServerFolder, encodedFilename)){
			logger.log(LogService.LOG_DEBUG, "Le fichier audio '"+encodedFilename+"' existe deje sur le serveur HTTPS");
		}
		else{
			logger.log(LogService.LOG_DEBUG, "Le fichier audio '"+encodedFilename+"' n'existe pas encore sur le serveur HTTPS");
			File encodedAudioFile = new File(audioTemporalyPath + encodedFilename);
			if(encodedAudioFile.exists()){
				logger.log(LogService.LOG_DEBUG, "Le fichier audio encode existe deje en local");
				download = true;
				encode = true;
			}
			else{
				downnloadAudioFile(fileName, filePath, fileUrl);
				download = true;
				
				// Encodage du fichier audio en vue de le pousser sur le serveur HTTPS
				logger.log(LogService.LOG_DEBUG, "Encodage du fichier audio :"+filePath+" (path : "+ configHttp.getString("audio_temp_path") +")");
				newAudioFile = PscsCoreUtil.encodeAudioFile(audioTemporalyPath, fileName, configAudio);
				encode = true;
			}
			
			// Recuperation des donnees du fichier audio encode
			newAudioFile.setBinary(recupDonneesAudio(newAudioFile.getName()));	
			
			// Upload du fichier audio encode sur le servuer HTTPS
			uploadAnnonceAudio(localServerFolder, newAudioFile);
		}
		
		return newAudioFile;
	}
	
	protected String getLocalServerFolder() {
		return "/pscs/";
	}


	/**
	 * Telecharge le fichier s'il n'est pas deje present dans le repertoire temporaire 
	 * @param fileName 
	 * @param fileUrl 
	 * @param filePath 
	 * 
	 * */
	public void downnloadAudioFile(String fileName, String destinationfilePath, String fileUrl) throws CoreException{
		File audioFile = new File(configHttp.getString("audio_temp_path") + fileName);
		// Le fichier audio originel existe en local sur la machine
		if(audioFile.exists()){
			logger.log(LogService.LOG_DEBUG, "Le fichier audio originel existe deje en local : '"+configHttp.getString("audio_temp_path") +fileName+"'");
		}
		// Le fichier audio originel n'existe pas en local sur la machine
		else{
			// Recuperation du fichier audio et enregistrement temporaire pour encodage
			byte[] fileData = downloadAudioFileFromHttpServer(fileUrl);
			PscsCoreUtil.saveAudioFile(destinationfilePath, fileData);
		}
	}


	public byte[] downloadAudioFileFromHttpServer(String fileUrl) throws CoreException {
		return HttpRequestFactory.getFileFromHttpServer(fileUrl);
	}
	
	public byte[] recupDonneesAudio(String filename) throws CoreException{
		try{
			File newFile = new File(configHttp.getString("audio_temp_path") + filename);
			FileInputStream fileInputStream = new FileInputStream(newFile);
			byte[] binaryFile = new byte[(int) newFile.length()];
			fileInputStream.read(binaryFile);
			logger.log(LogService.LOG_DEBUG, "Recuperation des donnees du fichier audio e envoyer: OK");
			fileInputStream.close();
			
			return binaryFile;
		} catch (IOException e) {
			throw new CoreException("Erreur lors de la recuperation des donnees du fichier audio e envoyer : " + configHttp.getString("audio_temp_path") + filename, e);
		}
	}
	
	public void uploadAnnonceAudio(String localServerFolder, AudioFile newAudioFile) throws IOException{
		logger.log(LogService.LOG_DEBUG, "Envoi du fichier audio sur le serveur HTTPS");
		FileUtils.postFile(localServerFolder, newAudioFile.getName(), newAudioFile.getBinary());
		logger.log(LogService.LOG_DEBUG, "Envoi du fichier audio sur le serveur HTTPS : OK");
	}	
	
	public IFluxTmlg envoiAnnonceAudio(IFluxTmlg flux, AudioFile newAudioFile) throws CoreException{
		String serverAudioEmbarqueURLGet =  configHttp.getString("url_embarque_server_get");
		
		// MAJ de l'annonce audio avec les infos du fichier audio sur le serveur https
		flux.getBody().getInfoVoyageur().getAnnonceAudio().setFileName(newAudioFile.getName());
		flux.getBody().getInfoVoyageur().getAnnonceAudio().setFormat(newAudioFile.getFormat());
		flux.getBody().getInfoVoyageur().getAnnonceAudio().setUrl(serverAudioEmbarqueURLGet + newAudioFile.getName());
		
		logger.log(LogService.LOG_DEBUG, "Envoi du message oBix e l'adaptateur: " + flux.toString());
		return flux;
	}
	
	/**
	 * Check if a audio file exists on the server.
	 * @param localServerFolder the url where to look for the file
	 * @param fileName the name of the file
	 * @return true if the file exists, else false
	 */
	public boolean verifAudioFileExistsOnServer(String localServerFolder, String fileName) {
		boolean isFileExist = false;
		if (localServerFolder != null){
			//Add a / at the end of the path if not present
			localServerFolder = FileUtils.checkPath(localServerFolder);
			isFileExist = FileUtils.isFileExist(localServerFolder + fileName);; 
		}
		
		return isFileExist; 
	}
	
}
