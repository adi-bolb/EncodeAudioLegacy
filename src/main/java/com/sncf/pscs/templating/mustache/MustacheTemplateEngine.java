package com.sncf.pscs.templating.mustache;

import java.io.IOException;
import java.io.StringWriter;
import java.util.Map;

import com.github.mustachejava.DefaultMustacheFactory;
import com.github.mustachejava.Mustache;
import com.sncf.pscs.templating.TemplateEngine;

public class MustacheTemplateEngine implements TemplateEngine {

	public String compile(String template, Map<String, String> vals) {
		Mustache mustache = new DefaultMustacheFactory().compile(template);
	    StringWriter writer = new StringWriter();
		try {
			mustache.execute(writer, vals ).flush();
		} catch (IOException e) {
			throw new RuntimeException(e);
		}
		return writer.toString();
	}

}
