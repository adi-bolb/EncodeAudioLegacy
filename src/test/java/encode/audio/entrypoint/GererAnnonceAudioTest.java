
package encode.audio.entrypoint;
import static org.mockito.Matchers.anyString;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.util.HashMap;
import java.util.Map;

import org.approvaltests.legacycode.LegacyApprovals;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;
import org.junit.runner.RunWith;
import org.mockito.invocation.InvocationOnMock;
import org.mockito.stubbing.Answer;
import org.powermock.api.mockito.PowerMockito;
import org.powermock.core.classloader.annotations.PrepareForTest;
import org.powermock.modules.junit4.PowerMockRunner;

import templating.TemplateEngine;
import templating.mustache.MustacheTemplateEngine;
import encode.audio.entrypoint.DataObject;
import encode.audio.entrypoint.FileUtils;
import encode.audio.entrypoint.GererAnnonceAudio;
import encode.audio.utils.CoreException;
import encode.audio.utils.Mp3Encoder;
import flux.ObixTmlgExeption;


@RunWith(PowerMockRunner.class)
@PrepareForTest(Mp3Encoder.class)
@UseReporter(MeldReporter.class)

public class GererAnnonceAudioTest {

	private static final Object[] ALL_BOOLEAN_VALUES = { true, false };

//	@Test
//	public void runOnce() throws Exception {
//		assertEquals("kbaskhjfksadjf", allGererAnnonceAudio(true, "wav", 128, "lame"));
//	}
	
	@Test 
	public void transformationLock() throws Exception {
		Object[] encodingActivated = ALL_BOOLEAN_VALUES;
		

		Object[] conversionBinaryName = {"lame"};
		Object[] bitrate = {128};
		Object[] audioExtension = {"mp3", "wav"};
		Object[] encoding = ALL_BOOLEAN_VALUES;
		Object[] downLoadError = ALL_BOOLEAN_VALUES;
		
		LegacyApprovals.LockDown(this, "allGererAnnonceAudioCatchExceptions", encodingActivated, audioExtension , bitrate, conversionBinaryName, encoding, downLoadError );
	}
	
	public String allGererAnnonceAudioCatchExceptions(Boolean encodingActivated, String audioExtension , Integer bitrate, String conversionBinaryName, Boolean encodingSuccess, Boolean downloadError){
		try {
			return allGererAnnonceAudio(encodingActivated, audioExtension, bitrate, conversionBinaryName, encodingSuccess, downloadError);
		}catch (Exception e) {
			e.printStackTrace();
			return e.getClass().getName();
		}		
	}
	
	public String allGererAnnonceAudio(Boolean encodingActivated, String audioExtension , Integer bitrate, String conversionBinaryName, Boolean encodingSuccess, Boolean downloadError)
			throws Exception {
		TemporaryFolder rootFolder = setup();
		try {
			
			GererAnnonceAudio aa = testableGererAnnonceAudio(rootFolder, downloadError);

			final File downloadDir = rootFolder.newFolder();
			downloadDir.mkdir();
			HttpDataObj httpDataObj = new HttpDataObj(downloadDir.getAbsolutePath() + "/", "http://localhost/get");
			DataObject configAudioTmp = new AudioDataObject(encodingActivated, audioExtension); // url_embarque_server_get,

			
			PowerMockito.mockStatic(Mp3Encoder.class);
			final String destinationFullPath = downloadDir.getAbsolutePath() + "/" + "10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3";
			final String resourceName = "test.mp3";

			Answer<Integer> encodingOperation = encodingSuccess ? successfullEncoding(resourceName, destinationFullPath) : encodingFailure();
			
			PowerMockito.when(Mp3Encoder.launchMp3Exec(anyString())).thenAnswer(encodingOperation);
			
			//When
			String flux = aa.demanderFlux(buildXmlRequest(audioExtension), configAudioTmp, httpDataObj);
			
			//Then
			return flux;
		}  finally {
			teardown(rootFolder);
			
		}
	}

	private Answer<Integer> successfullEncoding(final String sourceName,
			final String destinationFullPath) {
		return new Answer<Integer>() {

			public Integer answer(InvocationOnMock invocation)
					throws Throwable {
				String testFileName = getClass()
								.getClassLoader()
								.getResource(sourceName).getFile();
				FileUtils.copyFile(new File(testFileName).getAbsolutePath(), destinationFullPath);
				return 0;
			}
		};
	}
	
	private Answer<Integer> encodingFailure() {
		return new Answer<Integer>() {
			
			public Integer answer(InvocationOnMock invocation)
					throws Throwable {
				return 1;
			}
		};
	}

	private void teardown(TemporaryFolder rootFolder) {
		rootFolder.delete();
	}

	private TemporaryFolder setup() throws IOException {
		TemporaryFolder rootFolder = new TemporaryFolder();
		rootFolder.create();
		return rootFolder;
	}

	private String buildXmlRequest(final String audioExtension)
			throws ObixTmlgExeption {
		TemplateEngine te = new MustacheTemplateEngine();
		@SuppressWarnings("serial")
		Map<String, String> vals = new HashMap<String, String>() {{
			put("SOURCEFORMAT", audioExtension);
		}};
		String message = te.compile("src/test/resources/ogive2pscsAudio_with_transformed_audio.xml", vals );
		return message;
	}

	private GererAnnonceAudio testableGererAnnonceAudio(final TemporaryFolder rootFolder, final boolean downloadError) throws IOException {
		final File lclsrvrFolder = rootFolder.newFolder();
		lclsrvrFolder.mkdir();

		GererAnnonceAudio aa = new GererAnnonceAudio() {
			@Override
			public byte[] downloadAudioFileFromHttpServer(String fileUrl)
					throws CoreException {
				if (downloadError)
					throw new CoreException("cant download");
				else {
					try {
						InputStream resource = getClass()
								.getClassLoader()
								.getResourceAsStream("10.151.156.180Mon_Nov_04_140724_CET_2013343.wav");
						byte[] bts = new byte[1000000];
						resource.read(bts);
						return bts;
					} catch (Exception e) {
						throw new CoreException(e);
					}
				}
			}

			@Override
			protected String getLocalServerFolder() {

					return lclsrvrFolder.getAbsolutePath();
			}
		};
		return aa;
	}
}
