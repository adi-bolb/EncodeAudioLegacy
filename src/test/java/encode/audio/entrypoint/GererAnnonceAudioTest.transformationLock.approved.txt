[true, mp3, 128, lame, true, true] = encode.audio.entrypoint.AppTechnicalException 
[false, mp3, 128, lame, true, true] = encode.audio.entrypoint.AppTechnicalException 
[true, wav, 128, lame, true, true] = encode.audio.entrypoint.AppTechnicalException 
[false, wav, 128, lame, true, true] = encode.audio.entrypoint.AppTechnicalException 
[true, mp3, 128, lame, false, true] = encode.audio.entrypoint.AppTechnicalException 
[false, mp3, 128, lame, false, true] = encode.audio.entrypoint.AppTechnicalException 
[true, wav, 128, lame, false, true] = encode.audio.entrypoint.AppTechnicalException 
[false, wav, 128, lame, false, true] = encode.audio.entrypoint.AppTechnicalException 
[true, mp3, 128, lame, true, false] = FluxTmlg [body=AnnonceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[false, mp3, 128, lame, true, false] = FluxTmlg [body=AnnonceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[true, wav, 128, lame, true, false] = FluxTmlg [body=AnnonceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[false, wav, 128, lame, true, false] = FluxTmlg [body=AnnonceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.wav, format=wav, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.wav]] 
[true, mp3, 128, lame, false, false] = FluxTmlg [body=AnnonceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[false, mp3, 128, lame, false, false] = FluxTmlg [body=AnnonceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[true, wav, 128, lame, false, false] = encode.audio.entrypoint.AppTechnicalException 
[false, wav, 128, lame, false, false] = FluxTmlg [body=AnnonceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.wav, format=wav, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.wav]] 
